patroni/dcs.py

```diff
@@ -85,12 +86,39 @@ def conn_url(self):
         return self.member.conn_url
 
 
-class Failover(namedtuple('Failover', 'index,leader,member')):
-
+class Failover(namedtuple('Failover', 'index,leader,member,planned_at')):
+
+    """
+    >>> 'Failover' in str(Failover.from_node(1, '{"leader": "cluster_leader"}'))
+    True
+    >>> 'Failover' in str(Failover.from_node(1, '{"leader": "cluster_leader", "member": "cluster:member"}'))
+    True
+    >>> Failover.from_node(1, 'null') is None
+    True
+    >>> n = '{"leader": "cluster_leader", "member": "cluster:member", "planned_at": "2016-01-14T10:09:57.1394Z"}'
+    >>> 'tzinfo=' in str(Failover.from_node(1, n))
+    True
+    >>> Failover.from_node(1, None) is None
+    True
+    """
     @staticmethod
     def from_node(index, value):
-        t = [a.strip() for a in value.split(':')] + ['']
-        return Failover(index, t[0], t[1]) if t[0] or t[1] else None
+        if value is None:
+            return None
+
+        try:
+            data = json.loads(value)
+        except ValueError:
+            t = [a.strip() for a in value.split(':')] + ['']
+            return Failover(index, t[0], t[1], None) if t[0] or t[1] else None
```

Comment:
gotcha, this would indeed work, but is somewhat harder to read than the code that just checks the length of the list...